src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -61,40 +61,61 @@ public class DocumentValidator : IDocumentValidator
         };
 
         /// <inheritdoc/>
-        public async Task<IValidationResult> ValidateAsync(
-            string originalQuery,
+        public async Task<(IValidationResult validationResult, Variables variables)> ValidateAsync(
             ISchema schema,
             Document document,
+            VariableDefinitions variableDefinitions,
             IEnumerable<IValidationRule> rules = null,
             IDictionary<string, object> userContext = null,
             Inputs inputs = null)
         {
+            ValidationContext Initialize()
+            {
+                var context = System.Threading.Interlocked.Exchange(ref _reusableValidationContext, null) ?? new ValidationContext();
+                context.Schema = schema;
+                context.Document = document;
+                context.TypeInfo = new TypeInfo(schema);
+                context.UserContext = userContext;
+                context.Inputs = inputs;
+                return context;
+            }
+
             schema.Initialize();
 
+            ValidationContext context = null;
             bool success = false;
             bool useOnlyStandardRules = rules == null;
+
             if (useOnlyStandardRules)
             {
                 rules = CoreRules;
             }
             else if (!rules.Any())
             {
-                return SuccessfullyValidatedResult.Instance;
+                context = Initialize();
+                try
```

Comment:
I think the `finally` block is the same.  Can we move the `if (!rules.Any())` inside the `try` block and then eliminate the duplicate try..catch block and inline `MakeResult` so it reads easier?

Reply:
Done